### PR TITLE
walk up the chain to find a banner

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/2964 by making primary pages serve their own banner if they have one set, or "a parent's banner if there is one", walking up the parent chain until either there is a PrimaryPage in the parent chain with a banner set (and then we use that), or we hit the homepage (and then we use the default banner)

Test urls:
- top banner page with explicit banner set: https://foundation-s-inherit-pa-f5gtxv.herokuapp.com/en/campaigns/whats-this/ (Edit link: https://foundation-s-inherit-pa-f5gtxv.herokuapp.com/cms/pages/30/edit/)
- child banner page without explicit banner set: https://foundation-s-inherit-pa-f5gtxv.herokuapp.com/en/campaigns/success-drop-deep-general-run-break/child-without-banner/ (edit link: https://foundation-s-inherit-pa-f5gtxv.herokuapp.com/cms/pages/54/edit/)
- tertiary child, which I don't think we even use but it works anway: https://foundation-s-inherit-pa-f5gtxv.herokuapp.com/en/campaigns/success-drop-deep-general-run-break/child-without-banner/tetriary-page/ (edit link: https://foundation-s-inherit-pa-f5gtxv.herokuapp.com/cms/pages/55/edit/)

Also test url:
- top primary page without a banner set should use the default banner: https://foundation-s-inherit-pa-f5gtxv.herokuapp.com/en/campaigns/red-yet-cup-house-country-another/

### CODE REVIEW

**Comment by Pomax** [comment]
So, fun fact: this technically applies to all `PrimaryPages`, so it _might_ already work for mozfest!

**Comment by Pomax** [comment]
Ah, just checked, mozfest uses `class MozfestPrimaryPage(FoundationMetadataPageMixin, Page)`, so it doesn't directly extend `PrimaryPage`, so no free lunch... want to file the issue to get this working for Mozfest, too? Should be relatively easily achieved.

File: network-api/networkapi/wagtailpages/pagemodels/primary.py [hunk 1]
```diff
@@ -37,6 +37,21 @@ class PrimaryPage(FoundationMetadataPageMixin, Page):
         help_text='Choose an image that\'s bigger than 4032px x 1152px with aspect ratio 3.5:1',
     )
 
+    def get_banner(self):
+        if self.banner is None:
+            # note: we need to reverse get_ancestors, because by default
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/pagemodels/primary.py:42]: nice

**Comment by kristinashu** [comment]
Opened a follow up ticket for mozfest site https://github.com/mozilla/foundation.mozilla.org/issues/4789

**Pull request merged**